### PR TITLE
refactor: add logic to auth-modal

### PR DESCRIPTION
**Pull request by Zmicer-Yafremau**
#### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [x] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related trello card link

[_Describe the source of requirement, like related issue link._](https://trello.com/c/kUcZGeJ1/18-expand-registration-component)

#### 💡 Background and solution

Pass switching of tablets into auth-modal

#### 📷 Screenshots

-

### CODE REVIEW

File: src/app/components/auth-modal/auth-modal.component.html [hunk 1]
```diff
@@ -1 +1,20 @@
...
+        <mat-button-toggle class="header__button" value="signUp" #signUp>Sign up</mat-button-toggle>
+      </mat-button-toggle-group>
+    </header>
+    <div class="OAuth_wrapper">
+      <button mat-raised-button class="header__button OAuth__button google" id="googleButton">
+        Continue with Google
+      </button>
+      <button mat-raised-button class="header__button OAuth__button fb" id="googleButton">
+        Continue with Facebook
+      </button>
```
Comment (kubarenatioN) [inline src/app/components/auth-modal/auth-modal.component.html:16]: Is these buttons are part of `register` component? If so, then please, move them to appropriate place in `register` component

File: src/app/components/register/register.component.ts [hunk 2]
```diff
@@ -9,6 +9,11 @@ import { Validator } from 'src/app/shared/validators/validator';
   styleUrls: ['./register.component.scss'],
 })
 export class RegisterComponent implements OnInit {
+  
+  @Input() public signUpChecked!: boolean;
+
+  @Input() public loginChecked!: boolean;
```
Comment (kubarenatioN) [inline src/app/components/register/register.component.ts]: You dont need this flag here. This component is only for register logic

File: src/app/components/auth-modal/auth-modal.component.html [hunk 3]
```diff
@@ -1 +1,20 @@
...
+    </div>
+    <app-register [loginChecked]="login.checked" [signUpChecked]="signUp.checked"></app-register>
```
Comment (kubarenatioN) [inline src/app/components/auth-modal/auth-modal.component.html]: Look. Here you place `register` component. Also we need `login` near it. 
It's a simple - you just change the visibility of login/register by some enum value.
Placing login form inside `register` component is a mistake, so please move login form to `login` component.

**Review by kubarenatioN — CHANGES_REQUESTED** [review]
Better. Need some fixes of structure.

File: src/app/components/auth-modal/auth-modal.component.html (hunk 1 above)
Comment (Zmicer-Yafremau) [reply src/app/components/auth-modal/auth-modal.component.html:16]: Now it's common part

Comment (kubarenatioN) [reply src/app/components/auth-modal/auth-modal.component.html:16]: Is social buttons are common for login and register by design? If so, then never mind

Comment (Zmicer-Yafremau) [reply src/app/components/auth-modal/auth-modal.component.html:16]: yes

image

File: src/app/components/register/register.component.ts (hunk 2 above)
Comment (Zmicer-Yafremau) [reply src/app/components/register/register.component.ts]: done

File: src/app/components/auth-modal/auth-modal.component.html (hunk 3 above)
Comment (Zmicer-Yafremau) [reply src/app/components/auth-modal/auth-modal.component.html]: done

File: src/app/components/auth-modal/auth-modal.component.html [hunk 4]
```diff
@@ -1 +1,21 @@
...
+    <div class="OAuth_wrapper">
+      <button mat-raised-button class="header__button OAuth__button google" id="googleButton">
+        Continue with Google
+      </button>
+      <button mat-raised-button class="header__button OAuth__button fb" id="googleButton">
+        Continue with Facebook
+      </button>
+    </div>
+    <app-login *ngIf="login.checked" ></app-login>
+    <app-register *ngIf="signUp.checked"></app-register>
```
Comment (kubarenatioN) [inline src/app/components/auth-modal/auth-modal.component.html:19]: Even better would be create `enum authMode { login, register }`
But for now it's ok.

**Review by kubarenatioN — APPROVED** [review]
Great job 💯

**Review by LilithPrimary — APPROVED** [review]

**Review by ShArPman13 — APPROVED** [review]

**Pull request merged**